src/libCli/Completion.cpp

```diff
@@ -19,198 +19,242 @@ using namespace ArgParse;
 
 namespace cli
 {
-
-void printFishCompletions( std::vector<std::shared_ptr<ParsedElement> > & f_candidates, ParsedElement & f_parseTree, const std::string & f_args, bool f_debug)
-{
-    // completion requested :)
-    if(f_debug)
-    {
-        std::cerr << "Input string \"" << f_args << std::endl;
-    }
-
-    //size_t n = parseTree.getMatchedString().size();
-    size_t n = f_args.size();
-    for(auto candidate : f_candidates)
+    // This function modifies the candidate string, such that:
+    // - Only the arguments after the user input are counted as suggestion
+    // - Only one suggestion is completed at a a time
+    //   (everything after a whitespace in a suggestion should be anouther suggestion)
+    std::string getLastArgument(const std::string &f_candidateString, const char *f_delim, const std::string f_userInput, bool &f_isTrimmed)
     {
-        std::string candidateStr = candidate->getMatchedStringRaw();
+        std::string suggestion = f_candidateString;
+        int suggLength = suggestion.length();
+        std::size_t suggStart = f_userInput.find_last_of(f_delim);
 
-        if(f_debug)
+        // avoid substring error if suggStart (_pos) is undefined
+        if (suggStart >= suggestion.length() || suggStart == std::string::npos)
         {
-            std::cout << "------------------------------------------------" << std::endl;
-            std::cout << "candiate string: " << candidateStr << std::endl;
-            std::cout << "candiate's debug string: " << std::endl;
-            std::cout << candidate->getDebugString() << std::endl;
-            std::cout << "------------------------------------------------" << std::endl;
+            return (suggestion);
         }
 
-        std::string suggestion;
-        size_t start = n;
-        size_t end;
+        suggestion = suggestion.substr(suggStart, std::string::npos);
 
-        std::string suggestionDoc = candidate.get()->getShortDocument();
-        std::string suggestionDocRoot = f_parseTree.getShortDocument();
-        /// here the document (description of the tab completion) extracted from the original parsed tree and from the candidate tree should be compared,
-        /// if there is no new document of candidate in the parsed tree, that means the both extracted documents must same,
-        ///we only find the document at the right most position in the parsed tree (the newest) for tab completion.
-        if(suggestionDocRoot == suggestionDoc)
+        if (suggestion != " ")
```

Comment:
is this trimming step actually needed? I do not yet understand this :(
could you give an example where this is actually doing something.

Reply:
This trimming was implemented for testing purpose and not longer needed. I forgot to delete this.